### PR TITLE
Use an injected loader for better performance

### PR DESCRIPTION
**Pull request by apapirovski**
### What is accomplished by your PR?
Instead of using the nmf.afterResolve hook, use an injected loader for much faster performance on large projects. From 15s to 0.5s on a project with 10k+ files.

### Is there something controversial in your PR?
Modifying the original config although this seems to be relatively common judging by a quick audit of the ecosystem.

# Checklist

### New Feature / Bug Fix

- [x] Run unit tests to ensure all existing tests are still passing
- [ ] Add new passing unit tests to cover the code introduced by your PR

### CODE REVIEW

**Comment by woutervanvliet** [comment]
Do you have any documentation about this method?

**Comment by apapirovski** [comment]
@woutervanvliet what type of documentation are you looking for?

In terms of usage in the ecosystem, there are a number of plugins that do this or something similar:

https://github.com/vuejs/vue-loader/blob/64af07915ac47ebb6bad3223fd2bd15dee908196/lib/plugin.js
https://github.com/Esri/arcgis-webpack-plugin
https://github.com/faceyspacey/extract-css-chunks-webpack-plugin/blob/ae0f5b10e81326c7b7ec9f53337dd4a46a463cc2/src/index.js

The reason it's faster is that the current solution has to run the hook for each imported file which on large projects becomes prohibitively expensive.

**Comment by woutervanvliet** [comment]
@apapirovski That's pretty much what I was looking for. Just something from elsewhere that shows that others are using this method as well.

File: src/index.js [hunk 1]
```diff
@@ -62,21 +62,13 @@ export default class ServiceWorkerPlugin {
...
-
-        callback(null, result)
-      })
+    const data = JSON.stringify({
+      scriptURL: path.join(this.options.publicPath, this.options.filename),
     })
+    const loaderPath = `${path.join(__dirname, 'runtimeLoader.js')}?${data}`
+    const module = compiler.options.module
+    const rules = module.rules || module.loaders || (module.rules = [])
+    rules.push({ test: runtimePath, use: loaderPath })
```
[the remark this answers is not in the document]
Comment (apapirovski) [reply src/index.js]: I had to adjust slightly to account for `module.loaders` but changed.

**Comment by apapirovski** [comment]
@woutervanvliet @oliviertassinari anything else I can do here to help get this landed?

**Pull request merged**